src/main/java/com/uber/cadence/samples/hello/HelloPeriodic.java

```diff
@@ -24,131 +25,136 @@
 import com.uber.cadence.activity.Activity;
 import com.uber.cadence.activity.ActivityOptions;
 import com.uber.cadence.client.DuplicateWorkflowException;
-import com.uber.cadence.client.UntypedWorkflowStub;
 import com.uber.cadence.client.WorkflowClient;
 import com.uber.cadence.client.WorkflowException;
+import com.uber.cadence.client.WorkflowStub;
 import com.uber.cadence.worker.Worker;
 import com.uber.cadence.workflow.Workflow;
 import com.uber.cadence.workflow.WorkflowMethod;
 import java.time.Duration;
+import java.util.Optional;
 
 /**
- * Demonstrates a "cron" workflow that executes activity periodically.
- * Requires a local instance of Cadence server running.
+ * Demonstrates a "cron" workflow that executes activity periodically. Requires a local instance of
+ * Cadence server running.
  */
 public class HelloPeriodic {
 
-    private static final String TASK_LIST = "HelloPeriodic";
+  static final String TASK_LIST = "HelloPeriodic";
+  static final String PERIODIC_WORKFLOW_ID = "HelloPeriodic";
 
-    public interface GreetingWorkflow {
-        /**
-         * Use single fixed ID to ensure that there is at most one instance running.
-         * To run multiple instances  set different ids through
-         * WorkflowOptions passed to WorkflowClient.newWorkflowStub call.
-         */
-        @WorkflowMethod(
-                // At most one instance
-                workflowId = "HelloPeriodic",
-                // To allow starting workflow with the same ID after the previous one has terminated.
-                workflowIdReusePolicy = WorkflowIdReusePolicy.AllowDuplicate,
-                // Adjust this value to the maximum time workflow is expected to run
-                // It usually depends on number of repetitions and interval between them.
-                executionStartToCloseTimeoutSeconds = 300,
-                taskList = TASK_LIST)
-        void greetPeriodically(String name, Duration delay);
-    }
+  public interface GreetingWorkflow {
+    /**
+     * Use single fixed ID to ensure that there is at most one instance running. To run multiple
+     * instances set different ids through WorkflowOptions passed to WorkflowClient.newWorkflowStub
```

Comment:
ids -> IDs
to WorkflowClient.networkflowStub call -> to the WorkflowClient.networkflowStub call